### DISCUSSION TITLE
Berkeley QA Release 6f9d956

### DISCUSSION THREAD

**Discussion by deepthiskumar** [MinaProtocol/mina#11840] [Berkeley]
Mina's ecosystem partner, O(1) Labs, is continuously working on improvements to the core protocol and today we are excited to announce an updated unstable release `6f9d956` that is ready to test on the Berkeley QANet! Please report any issues in the comments on this post or tag any issues you create with `berkeley-rampup` and include berkeley `6f9d956` in the issue title.

### New Features/Improvements:

1. Accept transactions based on expected vs actual target nonces of zkapp transactions- https://github.com/MinaProtocol/mina/pull/11631

2. Report failures to include transactions in blocks- https://github.com/MinaProtocol/mina/pull/11373

3. Fixed issue reported by the Mina qa-task-force members- https://github.com/MinaProtocol/mina/issues/10879

4. Use Base64 for transaction ids- https://github.com/MinaProtocol/mina/issues/11505, https://github.com/MinaProtocol/mina/issues/11430
**Transaction ID**
The ID of a transaction is a base58Check encoding of itself. With zkapps, the transactions are bigger and using base58Check is impacting the client performance. Hence, proposing to change it from [[Base58Check](https://en.bitcoin.it/wiki/Base58Check_encoding)](https://en.bitcoin.it/wiki/Base58Check_encoding) to Base64.If this change is introduced, it will change the IDs of all the existing transactions.There are no changes to existing transaction types itself (Payments, Delegations, Fee transfers, and Coinbase) and only the encoding/decoding for the ID changes
**Transaction hash**
Transaction hash, as displayed in cli and graphQL commands, is currently computed by hashing transaction ID (which is a Base58Check encoding of a transaction). The new transaction hash for Payments, Delegations, Fee transfers, and Coinbase transactions will be now be the hash of the `Bin_prot` serialization of a transaction.
**Note**: ⚠️ All the cli and graphQL commands that uses ID have been updated to use base64 representation. Please ensure your integrations work with the new format

5. Rename Party → Account Update- https://github.com/MinaProtocol/mina/pull/11787

6. Specify libp2p-keypair explicitly- https://github.com/MinaProtocol/mina/pull/11764
**⚠️This change requires each node on every deployment to be configured with `--libp2p-keypair` argument ⚠️* *
  - Rename `-discovery-keypair` to `-libp2p-keypair`
  - Rename `mina advanced generate-libp2p-keypair` to `mina libp2p generate-keypair`
  - Make `-libp2p-keypair` a mandatory parameter
* Deployment strategy:
  - Rename `-discovery-keypair` to `-libp2p-keypair` whenever the former is used
  - For each node in every deployment generate a libp2p keypair with `mina libp2p generate-keypair`
    - Before generation, check if the key already exists
    - The goal is to reuse the same keypair across node restarts
  - Pass libp2p keypair to the node with `-libp2p-keypair`

7. Archive schema files changed. Please refer to the schema from the released commit
  - https://github.com/MinaProtocol/mina/blob/6f9d956890ef3f4beb5b4cb1a58a62352f33e5d1/src/app/archive/create_schema.sql
  - https://github.com/MinaProtocol/mina/blob/6f9d956890ef3f4beb5b4cb1a58a62352f33e5d1/src/app/archive/zkapp_tables.sql

SnarkyJS change log:  https://github.com/o1-labs/snarkyjs/blob/main/CHANGELOG.md

Note: zkApps Test Transaction CLI tool cannot be used to create token accounts in this release.

### Complete Changelog since `330d801` in:
Github Compare: https://github.com/MinaProtocol/mina/compare/330d8019f6c6884b44b0cdd0a9456d066cc04220...6f9d956890ef3f4beb5b4cb1a58a62352f33e5d1

### Debian Packages:
**Debian Repository:**
First, set up and update the `unstable` Debian Repository for your platform. Replace the word `CODENAME` with the appropriate codename for your machine, one of `bionic`, `focal`, `stretch`, `buster`, or `bullseye` and run:
```
echo "deb [trusted=yes] http://packages.o1test.net/ CODENAME unstable" | sudo tee /etc/apt/sources.list.d/mina-unstable.list
sudo apt-get update
```
Then, install the package(s) that you need:
**Mina Daemon:** `sudo apt-get install -y mina-berkeley=1.3.2beta2-release-2.0.0-6f9d956`
**zkApps Test Transaction CLI:** `sudo apt-get install -y mina-zkapp-test-transaction=1.3.2beta2-release-2.0.0-6f9d956`
[ zkApps Transaction CLI Documentation](https://github.com/MinaProtocol/mina/blob/ecbb3d490dce0f199894ebdf787a03b8fec64bd4/src/app/zkapp_test_transaction/README.md)

### Docker Images:
Every docker image is built for each of the supported platforms, replace `CODENAME` in the images below with your choice of `bionic`, `focal`, `stretch`, `buster`, or `bullseye` to select the base docker image.
**Mina Daemon:** `minaprotocol/mina-daemon:1.3.2beta2-release-2.0.0-6f9d956-CODENAME-berkeley`

### Step by Step Guide:
This version will only connect and function properly on the newly launched Berkeley QA Network. 
1. Create a libp2p keypair for a node the first time and persist it
`mina libp2p generate-keypair -privkey-path <path-to-the-key-file>`

2. Run mina daemon 
- Set environment variable `MINA_LIBP2P_PASS` with the password set for the libp2p keypair in step1
- Make sure to run the daemon with  `-libp2p-keypair <path-to-the-key-file>` and `--peer-list-url https://storage.googleapis.com/seed-lists/berkeley_seeds.txt`.

If you are running the correct version on the correct network **BERKELEY QA NET**, `mina client status` will show:
```
Chain id: 71114409c57753b12276a6518116b49d8a5d402c864b128b7bb6b2ef5b8b8853    
Git SHA-1:  6f9d956890ef3f4beb5b4cb1a58a62352f33e5d1

**Comment by garethtdavies** [comment]
If there are any issues with the libp2p options, it will just instantly crash and not print any error messages as to the specific issue. Make sure you have the `--libp2p-keypair` flag set, the `MINA_LIBP2P_PASS` env var, and permissions set correctly on the directory and libp2p key.

It would be useful for future releases to replicate the configuration errors that are printed e.g., on invalid block-producing keypair.

**Reply by nholland94** [reply]
> If there are any issues with the libp2p options, it will just instantly crash and not print any error messages as to the specific issue. Make sure you have the `--libp2p-keypair` flag set, the `MINA_LIBP2P_PASS` env var, and permissions set correctly on the directory and libp2p key.
> 
> It would be useful for future releases to replicate the configuration errors that are printed e.g., on invalid block-producing keypair.

Hi @garethtdavies!

Could you provide more information on how to reproduce this? I tried running the mina daemon both without the `--libp2p-keypair` flag, and with the `--libp2p-keypair` flag but without the `MINA_LIBP2P_PASS`. Both resulting in an immediate crash, but did contain the information of why it crashed.

```
# mina daemon   
Error parsing command line:

  missing required flag: --libp2p-keypair
```

```
# mina libp2p generate-keypair --privkey-path key/nathan
Password for new private key file: 
Again to confirm: 
libp2p keypair:
CAESQJHiVUhtJoOOwMrMYnSPW3zjcmVRAYzNWSt8dgpEUnwwUPly1DvtU53lb3mC9pF9zV1GgzVdQLfD0yww+/QdeYY=,CAESIFD5ctQ77VOd5W95gvaRfc1dRoM1XUC3w9MsMPv0HXmG,12D3KooWFGTQ3n8f5S3zKTY3ofu8SV4UEos7pRM9YBNrAh9ow8b7
root@cbdd2a6c69b2:~# mina daemon --seed --libp2p-keypair key/nathan
2022-09-19 17:49:41 UTC [Info] Mina daemon is booting up; built with commit "[DIRTY]6f9d956890ef3f4beb5b4cb1a58a62352f33e5d1" on branch "release/2.0.0"
2022-09-19 17:49:41 UTC [Info] Created daemon lockfile "/root/.mina-config/.mina-lock"
2022-09-19 17:49:41 UTC [Info] Registering async shutdown handler: "Remove daemon lockfile"
2022-09-19 17:49:41 UTC [Info] Daemon will expire at "2024-12-10 14:00:00-07:00"
2022-09-19 17:49:41 UTC [Info] Booting may take several seconds, please wait
2022-09-19 17:49:41 UTC [Info] Running async shutdown handler: "Remove daemon lockfile"
(monitor.ml.Error
 ( "\
  \nFATAL ERROR\
  \n\
  \n  \226\152\160  Mina encountered a configuration error loading libp2p keypair.\
  \n\
  \n  No password was specified in environment variable MINA_LIBP2P_PASS\
  \n")
 ("Raised at Mina_user_error.raise in file \"src/lib/mina_user_error/mina_user_error.ml\", line 6, characters 27-69"
  "Called from Async_kernel__Deferred0.bind.(fun) in file \"src/deferred0.ml\", line 54, characters 64-69"
  "Called from Async_kernel__Job_queue.run_job in file \"src/job_queue.ml\" (inlined), line 128, characters 2-5"
  "Called from Async_kernel__Job_queue.run_jobs in file \"src/job_queue.ml\", line 169, characters 6-47"))
```

The latter error could do with some better formatting, but it does still say the issue was that the `MINA_LIBP2P_PASS` env variable needs to be set.

**Reply by garethtdavies** [reply]
Ah, this is likely Docker image related. Here's the entire output with missing flags when using `minaprotocol/mina-daemon:1.3.2beta2-release-2.0.0-6f9d956-stretch-berkeley`.

```
[ERROR] Entrypoint script /entrypoint.d/. is not a regular file, ignoring
[ERROR] Entrypoint script /entrypoint.d/.. is not a regular file, ignoring
Mina process exited with status code 1
```

**Reply by lk86** [reply]
are you running with a log level that includes INFO? missing the stacktrace because of the docker entrypoint is not a big surprise but missing everything is strange

**Reply by garethtdavies** [reply]
Yes. Was basically the same setup as all prior versions, but had an issue with the libp2p permissions. There are a lot of other reports of users having the same issues and lack of logging these errors (also using Docker)
